### PR TITLE
fix: es6 style of Array inheritance to fix transpiler error

### PR DESCRIPTION
**Pull request by CodeNinjai**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

### CODE REVIEW

**Comment by CodeNinjai** [comment]
Travis CI brick: [ERR!] GitHub API: GitHub rate limit reached.

File: modules/swagger-codegen/src/main/resources/Javascript/es6/partial_model_generic.mustache [hunk 1]
```diff
@@ -6,7 +6,7 @@
 * @module {{#invokerPackage}}{{invokerPackage}}/{{/invokerPackage}}{{#modelPackage}}{{modelPackage}}/{{/modelPackage}}{{classname}}
 * @version {{projectVersion}}
 */{{/emitJSDoc}}
-export default class {{classname}} {
+export default class {{classname}}{{#parent}}{{^parentModel}}{{#vendorExtensions.x-isArray}} extends Array {{/vendorExtensions.x-isArray}}{{/parentModel}}{{/parent}}{
```
Comment (frol) [inline modules/swagger-codegen/src/main/resources/Javascript/es6/partial_model_generic.mustache]: Let's move the space before `extends` or after `Array` out of the "if" clause, so there is space between `classname` and `{` when there is no `extends Array`.

**Comment by frol** [comment]
Please, run `./bin/javascript-petstore-all.sh` to regenerate promise-es6 version as well.

File: samples/client/petstore/javascript-es6/src/model/AdditionalPropertiesClass.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ import ApiClient from '../ApiClient';
 * @module model/AdditionalPropertiesClass
 * @version 1.0.0
 */
-export default class AdditionalPropertiesClass {
+export default class AdditionalPropertiesClass{
```
Comment (frol) [inline samples/client/petstore/javascript-es6/src/model/AdditionalPropertiesClass.js]: Please, regenerate the clients one again to eliminate these changes.

Comment (CodeNinjai) [inline samples/client/petstore/javascript-es6/src/model/AdditionalPropertiesClass.js]: damn, sorry....

Comment (CodeNinjai) [inline samples/client/petstore/javascript-es6/src/model/AdditionalPropertiesClass.js]: Did, but forget to rebuild the codegen-cli ^^

**Comment by frol** [comment]
Looks good to me.

@CodeNinjai Thank you!

**Comment by wing328** [comment]
@frol thanks for reviewing the PR.

@CodeNinjai thanks for the fix 👍

**Pull request merged**